### ISSUE TITLE
Error when running "SELECT 1" query from AbstractDatabase in unsupported SQLDialect

### ISSUE THREAD

**Issue by johnkeates** [jOOQ/jOOQ#7248]
I can't seem to find a way to have jOOQ not bother with trying to find the schemata or catalog, both are not something the database seems to have. RazorSQL has no problem fetching database details, IntelliJ and DataGrip work too. They do display the database catalog and namespace as <unnamed> so they apparently are just blank. Not sure what select 1 "one" is supposed to do (select the first schema/namespace from catalog 1?), but FileMaker doesn't like it. Can I turn it off somehow?

### Expected behavior and actual behavior:
Running codegen with maven results in:
``` 
[ERROR] Could not load schemata  
org.jooq.exception.DataAccessException: SQL [select 1 "one"]; [FileMaker][FileMaker JDBC] FQL0001/(1:2): There is an error in the syntax of the query.
```

This query is apparently not doing anything for FileMaker over JDBC. Loading the driver in any DB browser works fine (shows all tables readable in the JDBC connection to the database).

### Steps to reproduce the problem:

Load `<driver>com.filemaker.jdbc.Driver</driver>` with `<name>org.jooq.util.jdbc.JDBCDatabase</name>` and it will fail.

### Versions:

- jOOQ: 3.10.5 (both plugin and dependency)
- Java: 1.8.0_131, vendor: Oracle Corporation
- Database (include vendor): FileMaker
- OS: "mac os x", version: "10.13.4", arch: "x86_64", family: "mac"
- JDBC Driver (include name if inofficial driver): com.filemaker.jdbc.Driver
- Apache Maven 3.5.0

**Comment by lukaseder** [comment]
Thanks for your report. We currently don't support the FileMaker database and there are currently no plans to do so.

We might be able to fix your specific issue if you are willing to investigate this further, though.

**Comment by johnkeates** [comment]
I'll investigate some more. So far it seems to get stuck on the fact that all catalogs and schemas are null because filemaker doesn't have any. I've added filemaker as a specific name next to the JDBC generic driver and removed everything it doesn't support, then added a default catalog and schema as placeholders, but I'm still stuck at the [select 1 "one"] query, I don't understand why it constructed it.

**Comment by lukaseder** [comment]
> I'm still stuck at the [select 1 "one"] query, I don't understand why it constructed it.

Hard to say without a stack trace, and other info on how to reproduce it :) Would it be possible to provide a minimal, complete, verifiable example?
https://stackoverflow.com/help/mcve

**Comment by johnkeates** [comment]
Not sure how, you'd need a Filemaker database so the JDBC connection is made, but after that you always get that query. I've tried setting breakpoints around the MetaImpl part where the getTables() is executed, gotten down to the DSL where it goes down a rabbit hole and between two method calls it suddenly has that query and wants to execute it.

A problem with Filemaker is that it's not really much of a database at all, it has SQL92 support (well, most of it..) but it only has TABLES, no views or anything, and only two virtual tables that describe what tables and columns exist, not much else (no sequences or functions or anything like that). It's comparable to MS Access in that way.

I've been looking into getting getTables0() to simply query the table description virtual table and use that, but it only wants Table instances with attached schema/catalog, which don't exist.

Up to this point, I've simply been working in jooq-meta and jooq from the 3.11-snapshot (master branch), and recompiling + running it to get some insight in the jooq architecture, some databases also seem to lack most basic RDBMS features so it shouldn't be much of a stretch to get Filemaker to work. 

Doing manual queries works, reading the two virtual tables to get metadata works, and for the my current project, I only have to interact with existing tables, so the lack of table creation/modification isn't much of a hurdle right now.

**Comment by lukaseder** [comment]
John, thanks for explaining :) But how can I help?

**Comment by johnkeates** [comment]
Let's start with this; what type of stack trace would you need? A debug run with full output? (also, I use IntelliJ but I can run from a shell too if that makes a difference). Also, how can I influence the generator process, i.e. make it skip catalogs and schema selection and just get tables right away? getTables0() seems end up making calls to getTables() no matter what.

**Comment by johnkeates** [comment]
This is the (somewhat small?) stacktrace that comes with the error when executed using maven:
```
[ERROR] Error while fetching tables
org.jooq.exception.DataAccessException: Error while running ConnectionCallable
	at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:635)
	at org.jooq.impl.MetaImpl.meta(MetaImpl.java:160)
	at org.jooq.impl.MetaImpl.access$200(MetaImpl.java:135)
	at org.jooq.impl.MetaImpl$MetaSchema.getTables(MetaImpl.java:331)
	at org.jooq.util.filemaker.FilemakerDatabase.getTables0(FilemakerDatabase.java:185)
	at org.jooq.util.AbstractDatabase.getTables(AbstractDatabase.java:1208)
	at org.jooq.util.JavaGenerator.generateSchemaIfEmpty(JavaGenerator.java:393)
	at org.jooq.util.JavaGenerator.generateCatalogIfEmpty(JavaGenerator.java:379)
	at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:353)
	at org.jooq.util.GenerationTool.run(GenerationTool.java:735)
	at org.jooq.util.GenerationTool.generate(GenerationTool.java:215)
	at org.jooq.util.maven.Plugin.execute(Plugin.java:195)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.jooq.exception.DataAccessException: SQL [select 1 "one"]; [FileMaker][FileMaker JDBC] FQL0001/(1:2): There is an error in the syntax of the query.
	at org.jooq_3.11.0.DEFAULT.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2290)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:730)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:363)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:320)
	at org.jooq.impl.SelectImpl.fetch(SelectImpl.java:2641)
	at org.jooq.util.AbstractDatabase$1.start(AbstractDatabase.java:253)
	at org.jooq.impl.ExecuteListeners.start(ExecuteListeners.java:130)
	at org.jooq.impl.ExecuteListeners.<init>(ExecuteListeners.java:124)
	at org.jooq.impl.ExecuteListeners.get(ExecuteListeners.java:83)
	at org.jooq.impl.DefaultDSLContext.fetchLazy(DefaultDSLContext.java:1266)
	at org.jooq.impl.DefaultDSLContext.fetchLazy(DefaultDSLContext.java:1283)
	at org.jooq.impl.DefaultDSLContext.fetch(DefaultDSLContext.java:1121)
	at org.jooq.impl.MetaImpl$MetaSchema$1.run(MetaImpl.java:365)
	at org.jooq.impl.MetaImpl$1.run(MetaImpl.java:163)
	at org.jooq.impl.MetaImpl$1.run(MetaImpl.java:160)
	at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:632)
	... 34 more
Caused by: com.filemaker.jdbc.FMSQLException: [FileMaker][FileMaker JDBC] FQL0001/(1:2): There is an error in the syntax of the query.
	at com.filemaker.jdbc.FM_API.prepare(Unknown Source)
	at com.filemaker.jdbc.FM_API.prepareRS(Unknown Source)
	at com.filemaker.jdbc.FM_API.prepareRS(Unknown Source)
	at com.filemaker.jdbc1.CommonJ1Statement.<init>(Unknown Source)
	at com.filemaker.jdbc2.CommonJ2Statement.<init>(Unknown Source)
	at com.filemaker.jdbc3.CommonJ3Statement.<init>(Unknown Source)
	at com.filemaker.jdbc3.J3PreparedStatement.<init>(Unknown Source)
	at com.filemaker.jdbc3.J3Connection.prepareStatement(Unknown Source)
	at com.filemaker.jdbc2.CommonJ2Connection.prepareStatement(Unknown Source)
	at org.jooq.impl.ProviderEnabledConnection.prepareStatement(ProviderEnabledConnection.java:109)
	at org.jooq.impl.SettingsEnabledConnection.prepareStatement(SettingsEnabledConnection.java:73)
	at org.jooq.impl.AbstractResultQuery.prepare(AbstractResultQuery.java:236)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:321)
	... 47 more
```

I'll implement org.jooq.util.Database and see if I can make the broken database behave a bit better.

**Comment by lukaseder** [comment]
Oh cool, thanks a lot. That definitely helps tracking down the issue. I completely forgot about that query. The query is run to have a first "contact" with the database (including the class loading this might incur), which prevents some other performance related logs from happening. See: #4974

But of course, if the dialect is not supported, that query might not work, so we should catch any exception that is thrown from this query and ignore it.

That should be an easy fix.

**Comment by lukaseder** [comment]
... until the fix ships, I don't think there's a workaround short of either re-implementing the entire `AbstractDatabase` on your side, or patching it with the fix.

**Comment by lukaseder** [comment]
Fix will be ported to 3.9.7 (#7255) and 3.10.6 (#7254)

**Comment by lukaseder** [comment]
Creating your own dialect is one option, but I suggest you try using the HSQLDB dialect instead and see how far you get. This has worked decently well for people working with HiveSQL in the past, for instance. HSQLDB is a rather standards compliant dialect. As I mentioned before, while it's not too difficult to create the code generator support (perhaps 3-5 days worth of effort to cover all edge cases), adding dialect support takes much longer, and you will have to patch your version all the time when you upgrade jOOQ as we frequently re-organise the internals.

Having said so, while FileMaker is almost as popular as MS Access *generally* (e.g. according to this ranking: https://db-engines.com/en/ranking), it isn't really that popular in the Java ecosystem, given the comparable alternatives including SQLite, H2, HSQLDB, Derby. I have added a feature request to the roadmap (#7260), but given the fact that you're the first person having asked for an integration in almost 10 years of jOOQ's existence, I'm afraid we can't prioritise such an integration right now :-)
